### PR TITLE
feat: extract axon-tools, add serialization

### PR DESCRIPTION
**Pull request by wenyuanhust**
## What this PR does / why we need it?

This PR 
- extract forcerelay related stuff into dir devtools/axon-tools
- add deserialization of Block, Proposal .etc for forcerelay

### What is the impact of this PR?

No Breaking Change

**PR relation**:
- Ref #

<details><summary>CI Settings</summary><br/>

### **CI Usage**

**Tip**: Check the CI you want to run below, and then comment `/run-ci`.

**CI Switch**

- [ ] Web3 Compatible Tests
- [ ] OCT 1-5 And 12-15
- [ ] OCT 6-10
- [ ] OCT 11
- [ ] OCT 16-19
- [ ] v3 Core Tests

### **CI Description**

| CI Name                                   | Description                                                               |
| ----------------------------------------- | ------------------------------------------------------------------------- |
| *Web3 Compatible Test*                    | Test the Web3 compatibility of Axon                                       |
| *v3 Core Test*                            | Run the compatibility tests provided by Uniswap V3                        |
| *OCT 1-5 \| 6-10 \| 11 \| 12-15 \| 16-19* | Run the compatibility tests provided by OpenZeppelin                      |

</details>

### CODE REVIEW

File: devtools/axon-tools/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+name = "axon-tools"
+version = "0.1.1"
+edition = "2021"
+authors = ["Axon Dev <axon@axonweb3.io>"]
+license = "MIT"
+include = ["src/*", "README.md", "LICENSE"]
+readme = "README.md"
+keywords = ["axon", "tool"]
+categories = ["cryptography"]
+repository = "https://github.com/axonweb3/axon-tools"
```
Comment (KaoImin) [inline devtools/axon-tools/Cargo.toml]:
```suggestion
repository = "https://github.com/axonweb3/axon"
```

File: protocol/src/codec/mod.rs [hunk 2]
```diff
@@ -109,6 +109,72 @@ where
...
+        Err(serde::de::Error::custom("Invalid format"))
+    }
+}
+
+pub fn deserialize_hex_u64<'de, D>(deserializer: D) -> Result<u64, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s = String::deserialize(deserializer)?;
+    // println!("deserialize s: {}", s);
```
Comment (KaoImin) [inline protocol/src/codec/mod.rs]: These `println!` should be removed.

File: protocol/src/types/block.rs [hunk 3]
```diff
@@ -53,17 +54,33 @@ pub struct Proposal {
     pub prev_state_root:          MerkleRoot,
     pub transactions_root:        MerkleRoot,
     pub signed_txs_hash:          Hash,
-    #[cfg_attr(feature = "hex-serialize", serde(serialize_with = "serialize_uint"))]
+    #[cfg_attr(feature = "hex-serialize", serde(serialize_with = "serialize_uint",))]
+    #[cfg_attr(
```
Comment (KaoImin) [inline protocol/src/types/block.rs:58]: I do not understand the necessity of derive custom `Deserialize` function.

**Review by KaoImin — DISMISSED** [review]

**Review by KaoImin — CHANGES_REQUESTED** [review]
👆 questions should be resolved

File: protocol/src/types/block.rs (hunk 3 above)
Comment (wenyuanhust) [reply protocol/src/types/block.rs:58]: forcerelay will call methods like get_block_by_id, the datas gotten should be deserialized to construct correct data structures

**Review by Flouse — CHANGES_REQUESTED** [review]
Please fix the CI checks.
image

**Pull request closed without merging**